File: js/src/types/ohttp.ts
import type * as v from 'valibot';
import type { OhttpAttestationSchema } from '../schemas';
import type { SigningIdentity } from './attestation-common';

/** Wire-format OHTTP proof in Gateway and direct attestation reports. */
export type OhttpAttestationResponse = v.InferOutput<
  typeof OhttpAttestationSchema
>;

/** Signed OHTTP key configuration advertised by an attestation endpoint. */
export type OhttpAttestation = {
  readonly signingAlgo: 'ed25519';
  readonly signingKey: string;
  readonly keyConfig: string;
  readonly signature: string;
};

export type VerifyOhttpKeyConfigParams = {
  readonly ohttpAttestation: OhttpAttestation;
  /** Identity obtained from a successfully verified Gateway or direct model attestation. */
  readonly signer: SigningIdentity;
};

/** Configure OHTTP transport with a previously authenticated key configuration. */
export type CreateOhttpFetchParams = {
  readonly keyConfig: Uint8Array;
  readonly baseUrl: string;
  readonly fetch?: typeof globalThis.fetch;
  /** Additional inner header names to expose on the outer request. Authorization is always forwarded. */
  readonly forwardedHeaders?: readonly string[];
};

File: py/src/nearai_inference_sdk/types/ohttp.py
"""Signed OHTTP key configuration advertised by Gateway evidence."""

from __future__ import annotations

from dataclasses import dataclass
from typing import Literal


@dataclass(frozen=True, kw_only=True)
class OhttpAttestation:
    """Hexadecimal key material and a signature over raw configuration bytes."""

    signing_algo: Literal['ed25519']
    signing_key: str
    key_config: str
    signature: str
